enhancements/baremetal-observability/README.md

```diff
@@ -0,0 +1,218 @@
+
+## Summary
+
+We want a way to gather bare metal metrics from tenant clusters and have them
+available at some configurable endpoint for the cloud provider to manage. This
+feature will be easily toggleable per tenant cluster.
+
+## Motivation
+
+Gathering baremetal metrics is an important aspect of observability in cloud
+systems. It allows the cloud provider to make informed decisions on cost
+estimates and abnormal hardware behavior.
+
+If the cloud provider decides to expose baremetal metrics to tenants,
+it would allow them to also make informed decisions on cost estimates
+and essential hardware metrics if they should use it for their projects.
+
+### User Stories
+
+* As a provider, I want to easily deploy and undeploy bare metal
+  observability using an automation application like Ansible
+  or an operator.
+* As a provider, I want to be able to access a metric endpoint so I
+  can connect it to Prometheus compatible frontend applications and
+  stacks.
+* As a provider, I want my metrics to come with labels about the
+  originating clusters and nodes so I can build my own RBAC system
+  to the metrics for the tenants.
+* As a provider, I want to easily include another metric exporter if
+  I decide I want to use something other than Prometheus's IPMI and
+  SNMP exporters.
+
+### Goals
+
+This will be a success if this can be deployed as an optional feature
+in the OSAC installation process. The implementation will rely on ESI
+and OpenStack for retrieving information on where to get baremetal
+metrics. This will be using Prometheus compatible applications
+because the Prometheus is built-in to OpenShift.
+
+This will be a success if there is minimal to no communication needed
+between the tenant and the provider regarding additional observability
+management and configuration.
+
+### Non-Goals
+
+* We will not be looking into how to use metrics for billing purposes.
+* We will not be looking into proxies/applications that would usually
+  go on top of the metric API endpoint.
+* We will not be looking into collecting from in-OS exporters in bare
+  metal clusters. This may call for a later enhancement.
+* We will not be looking into fine-grain access to metrics, but we
+  can implement metric labeling with cluster and node labels now to
+  help with implementing fine-grain access later. This will call for
+  a later enhancement.
+
+## Proposal
+
+The proposed implementation makes use of the [multi-target exporter pattern](https://prometheus.io/docs/guides/multi-target-exporter/#the-multi-target-exporter-pattern),
+notably the [IPMI-exporter](https://github.com/prometheus-community/ipmi_exporter)
+and the [SNMP-exporter](https://github.com/prometheus/snmp_exporter).
+Multi-target exporters have properties perfect for our goals:
+* the exporter does not have to run on the machine the metrics are taken from
+* the exporter will get the target’s metrics via a network protocol
+* the exporter can query multiple targets
```

Comment:
Is there a target for the metrics exported by the exporter - for example, ACM Hub Prometheus? Or is the idea that the cloud provider can configure where the exporter feeds its metrics?